packages/coinstac-ui/app/render/state/ducks/auth.js

```diff
@@ -145,28 +157,30 @@ export const signUp = applyAsyncLoading(user => (dispatch, getState) => axios.po
     return initCoreAndSetToken(user, data, appDirectory, dispatch);
   })
   .catch((err) => {
-    if (err.response && err.response.data && (err.response.data.message === 'Username taken'
-          || err.response.data.message === 'Email taken')) {
-      dispatch(setUser({ ...INITIAL_STATE, error: err.response.data.message }));
+    const { statusCode, message } = getErrorDetail(err);
+    if (statusCode === 409) {
```

Comment:
Did you test this? Is the API already returning 409 in these cases? Maybe you'll need to make a change on the API.